docs/scos/dev/front-end-development/202307.0/oryx/technology.md

```diff
@@ -0,0 +1,77 @@
+- [Vite](#vite)
+- [Storybook](#storybook)
+- [Web hosting](#web-hosting)
+
+
+## HTML
+
+HTML is a markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain. Also, Oryx prioritizes accessibility principles ensuring access to information and functionality for all users.
+
+## CSS
+
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML.
+
+Oryx uses standard CSS syntax and does not use CSS preprocessors because Oryx does not use a global CSS utility library; instead, we have isolated styles per component and because of that we do not want to load global css utils in each component.
+
+Also unlike some other CSS methodologies, such as BEM (Block, Element, Modifier), Oryx does not rely on a specific naming convention to structure its CSS classes.
+
+## TypeScript
+TypeScript is a superset of JS that adds optional static typing, classes, and interfaces to the language. In Oryx, TypeScript is used to provide type safety and better code organization. It allows developers to catch errors early in the development process and provides better code navigation and refactoring capabilities.
+
+## Web Components
+In Oryx, Web Components are used to create reusable UI components that can be used across the application. It allows developers to create complex components with minimal code duplication and provides better code organization and maintainability. At the same time it means Oryx component are compatible with any other frontend framework and can be integrated into almost any web-application.
+
+## Lit
+Lit is a lightweight library for creating web components using TypeScript and HTML templates. In Oryx, Lit is used to create reusable and encapsulated UI components that can be easily styled and extended. It provides a simple and intuitive API for creating custom elements and allows developers to write less boilerplate code.
+You can use Lit to customise Oryx, but you can also use another framework to build web components. Using Lit will give you a few advantages:
+- Reuse Component Mixins and controllers.
+- Out of the box integration with optimised (reactivity)[https://docs.spryker.com/docs/scos/dev/front-end-development/202212.0/oryx/reactivity/reactivity.html] and partial hydration concepts.
+- Out of the box integration with SSR.
+
+## RxJS
+RxJS is a reactive programming library for JS. In Oryx, RxJS is used to handle asynchronous operations. It provides a declarative way to handle events and data streams, making it easier to manage complex data flows and state transitions.
+RxJS is predominantly used in the service layer to provide a reactive system that emit new values while you navigate through the (SPA) experience.
+RxJS is not a main citizen in the component logic, as we try to hide the reactivity altogether to avoid complexity
```

Comment:
What is the main citizen? :D

Reply:
It means that we prefer to avoid rxjs usage inside component. We can rephrase it.